common/src/main/java/com/wynntils/gui/render/MapRenderer.java

```diff
@@ -99,10 +116,226 @@ public static void renderMapQuad(
         bufferBuilder.end();
         BufferUploader.end(bufferBuilder);
 
+        float mapTopX = centerX - halfRenderedWidth;
+        float mapLeftZ = centerZ - halfRenderedHeight;
+        float mapBottomX = centerX + halfRenderedWidth;
+        float mapRightZ = centerZ + halfRenderedHeight;
+
+        List<ServicePoi> servicePois = MapModel.getAllPois().stream()
+                .filter(poi -> {
+                    if (!(poi instanceof ServicePoi)) return false;
+
+                    return isPoiVisible(
+                            mapCenterX,
+                            mapCenterZ,
+                            centerX,
+                            centerZ,
+                            scale,
+                            mapTopX,
+                            mapLeftZ,
+                            mapBottomX,
+                            mapRightZ,
+                            poi);
+                })
+                .map(poi -> (ServicePoi) poi)
+                .toList();
+
+        if (renderMapLabels) {
+            List<LabelPoi> labelPois = MapModel.getAllPois().stream()
+                    .filter(poi -> poi instanceof LabelPoi)
+                    .map(poi -> (LabelPoi) poi)
+                    .toList();
+
+            for (LabelPoi labelPoi : labelPois) {
+                float x = labelPoi.getLocation().getX();
+                float z = labelPoi.getLocation().getZ();
+                double distanceX = x - mapCenterX;
+                double distanceZ = z - mapCenterZ;
+
+                float textureXPosition = (float) (centerX
+                        + distanceX / scale
+                        - FontRenderer.getInstance()
+                                        .getFont()
+                                        .width(labelPoi.getLabel().getName())
+                                / 2f
+                                / scale);
+                float textureZPosition = (float) (centerZ + distanceZ / scale - 4.5f / scale);
```

Comment:
Can't you just change the alignment configs

Reply:
Scale needs to be factored in, font renderer can't do that.